apps/core/lib/core/accounts.ex

```diff
@@ -19,13 +20,30 @@ defmodule Core.Accounts do
   def user(id) when is_integer(id), do: Repo.get_record(User, id)
 
   @spec create_user(attrs :: map()) :: Repo.modify_result_t(User.t())
-  def create_user(%{} = attrs), do: User.create_changeset(attrs) |> Repo.insert()
+  def create_user(%{} = attrs) do
+    with {:ok, user} <- User.create_changeset(attrs) |> Repo.insert(),
+         :ok <-
+           EventRepo.save_event(
+             :user_created,
+             {user.id, Map.delete(attrs, "password"), user.inserted_at}
+           ) do
+      {:ok, user}
+    end
+  end
 
   @spec update_user(User.t(), attrs :: map()) :: Repo.modify_result_t(User.t())
   def update_user(%User{} = user, %{} = attrs) do
-    User.update_changeset(user, attrs) |> Repo.update()
+    with {:ok, user} <- User.update_changeset(user, attrs) |> Repo.update(),
+         :ok <- EventRepo.save_event(:user_updated, {user.id, attrs, user.updated_at}) do
```

Comment:
Filtering `password` while creating doesn't work, but if it would, the behavior would be inconsistent with update here.

Reply:
Also, it is not a big deal, but attributes for the update could just include changeset's changes.